### ISSUE TITLE
Create a snap package (the hidden files or folder accees of ~/.steam/ can now be solved!)

### ISSUE THREAD

**Issue by heidiwenger** [DavidoTek/ProtonUp-Qt#123]
On Ubuntu, there is no .deb package and putting up flatpak support takes terminal commands and too much effort for a new comer to Linux (and Ubuntu is the main destination).  

Referring to this partly solved issue (a flatpak was created but snap had a hurdle) https://github.com/DavidoTek/ProtonUp-Qt/issues/3

The technic that provides the same thing (to access those needed files / folders) as in the flatpack package is: https://snapcraft.io/docs/personal-files-interface

"The personal-files interface provides access to the specified files in the user’s home. This interface gives privileged access to the user’s data.

[Auto-connect](https://snapcraft.io/docs/interface-management#heading--auto-connections): no
[Super-privileged](https://snapcraft.io/docs/auto-connection-mechanism#heading--super): yes "

**Comment by DavidoTek** [comment]
> putting up flatpak support takes terminal commands and too much effort for a new comer to Linux (and Ubuntu is the main destination).

Yeah, it's probably easier to connect the personal files interface than installing Flatpak, even though you probably should do that as many apps are available as Flatpak.

ProtonUp-Qt could detect if it is running as Snap and if the personal files interface is connected. If that is not the case it can show the user a message like `You are running the Snap version. You either need to connect the Personal Files Interface [or use the Flatpak version]. Please open a Terminal and enter snap connect blahblah`.

I will take a look at creating a Snap package, if anyone wants to contribute a `snapcraft.yaml` in the meantime, feel free to do so.

**Comment by heidiwenger** [comment]
> > putting up flatpak support takes terminal commands and too much effort for a new comer to Linux (and Ubuntu is the main destination).
> 
> Yeah, it's probably easier to connect the personal files interface than installing Flatpak, even though you probably should do that as many apps are available as Flatpak.
> 
> ProtonUp-Qt could detect if it is running as Snap and if the personal files interface is connected. If that is not the case it can show the user a message like `You are running the Snap version. You either need to connect the Personal Files Interface [or use the Flatpak version]. Please open a Terminal and enter snap connect blahblah`.
> 
> I will take a look at creating a Snap package, if anyone wants to contribute a `snapcraft.yaml` in the meantime, feel free to do so.

Amazing stuff. All kinds of aid for this can be gotten from https://forum.snapcraft.io/t/protonup-qt-to-easily-install-proton-ge-for-steam/26492/2 additionally.

**Comment by vrement** [comment]
Came for the same. There's no snap package. Is there any progress for one yet?

**Comment by sonic2kk** [comment]
As far as I know, this has not been picked up. I didn't see anything in the commit history or any PRs directly related to this. I'm also not intimately familiar with Snap packaging (or really, any application packaging) so I'm not sure what needs to be done to implement this.

Probably a good place to start would be creating a `snapcraft.yaml` file, which is likely some kind of manifest file detailing an application's dependencies and permissions. I'm unsure of how to structure the dependencies, or of the overall format of the file, but a good reference might be the [Flatpak JSON manifest](https://github.com/flathub/net.davidotek.pupgui2/blob/master/net.davidotek.pupgui2.json). This should give an idea of what types of permissions ProtonUp-Qt needs (such as access to devices for controller support) as well as which files/folders it needs access to, and what kind of read/write permissions it needs (for example, it needs create permissions for `~/stl`, and read-only permissions for a bunch of Lutris folders).

**Comment by DavidoTek** [comment]
I think following steps need to be takes to create a Snap package:

- Create a `snapcraft.yaml` as sonic2kk already pointed out
   - Putting that in a separate repo is probably cleaner, I'm not sure if it's possible to only trigger builds on new releases (tags)
- Request [`classic` confinement](https://snapcraft.io/docs/snap-confinement) from Canonical: I don't think we can access the mandatory files without that. The alternative would be for users to manually grant permissions to all folders, I don't think that is viable.
- Create a better CI/CD pipeline: Ideally, a new release should not only build an AppImage, but also create a PR to the Flatpak (and later Snap) repo that contains 1) the link to latest release 2) updated Python dependencies based on `requirements.txt`. That would keep the needed maintenance low.

Maintaining the Snap would require following:

- Push commits to the main branch of the Snap repo (that will automatically trigger new builds)
- Maybe rebuilding if there are any security patches available
- Testing the builds and pushing them from edge to beta/stable.

I haven't planned to do that at the moment, but I'm happy to receive PRs or if anyone would like to maintain a snap package.

**Comment by vrement** [comment]
Alright, well i hope someone would pick this up then. I'll check back later :)

**Comment by vrement** [comment]
It has gone about 4 months now and i remembered this one. If there is anything to report i'd love to hear! I think ProtonUP is one of crucial tools for gamers. Would be great if it was available easily on Ubuntu as the LTS comes already nexth month.

**Comment by sonic2kk** [comment]
To my knowledge, no one has offered to create a Snap. At least they haven't posted about it here or made any PRs for a Snapcraft yaml or any CI/CD changes.

**Comment by DavidoTek** [comment]
This looks promising: https://discourse.ubuntu.com/t/ubuntu-desktop-s-24-10-dev-cycle-part-5-introducing-permissions-prompting/47963

> How prompting gives users control over the home interface
> [The home interface 4](https://forum.snapcraft.io/t/the-home-interface/7838) is a common interface required by desktop applications as it allows access to non-hidden files owned by the user in their home directory

Hmm, sounds like only non-hidden folders can be selected. Let's see.
(It could still work for Steam snap when installed in `~/snap` as this is not a hidden folder. That will likely change however with the introduction of the hidden snap folder `~/.snap` https://github.com/DavidoTek/ProtonUp-Qt/issues/445)

**Comment by vrement** [comment]
> This looks promising: https://discourse.ubuntu.com/t/ubuntu-desktop-s-24-10-dev-cycle-part-5-introducing-permissions-prompting/47963
> 
> > How prompting gives users control over the home interface
> > [The home interface 4](https://forum.snapcraft.io/t/the-home-interface/7838) is a common interface required by desktop applications as it allows access to non-hidden files owned by the user in their home directory
> 
> Hmm, sounds like only non-hidden folders can be selected. Let's see. (It could still work for Steam snap when installed in `~/snap` as this is not a hidden folder. That will likely change however with the introduction of the hidden snap folder `~/.snap` #445)

Came to request for a snap package too. Is there any news on the technical possibility on it? Looks like its now doable if someone packages?

**Comment by sonic2kk** [comment]
To my knowledge no work has been started on this (not a maintainer, I just contribute every so often, so I could be wrong).

I believe this is technically feasible, and that having someone do the legwork of packaging (including setting up a pipeline for building it like we have for AppImage) is the actual work-effort.
